app/Community/Policies/MessageThreadPolicy.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Community\Policies;
+
+use App\Community\Models\MessageThread;
+use App\Site\Models\User;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class MessageThreadPolicy
+{
+    use HandlesAuthorization;
+
+    public function view(User $user, MessageThread $message): bool
+    {
+        return true;
```

Comment:
This should check if a given user is allowed to view the thread (same for delete).
Otherwise it's possible to view threads that one is not a participant of?

Reply:
The participant restriction is currently enforced in `MessageThreadController::show`. I can't actually get these methods to be called - I'm guessing it's related to the route stuff.

Reply:
I'm clearly still missing something.
```
    public function create(Request $request): View
    {
        $this->authorize('create', Message::class);
```
And I put a divide by zero exception in MessagePolicy::create, and it doesn't fail.

Reply:
Yep. That seems to have been my problem.

If admins circumvent the policies, we can't really put the participation logic into the Policy file, or admins could view any message thread. Do you still want me to put authorize calls in every method, but leave the participation check where it is?